lib/upload.js

```diff
@@ -242,6 +272,16 @@ class Upload {
       clearTimeout(this._retryTimeout);
       this._retryTimeout = null;
     }
+
+    if (shouldTerminate) {
+      Upload.terminate(this.url, this.options, (err, xhr) => {
+        if (err || !this._hasStorage()) {
```

Comment:
Can you explain what the purpose of `|| !this._hasStorage()` here is?

Reply:
I have refactored the code so the intention is clearer